### ISSUE TITLE
add_text not working unless closing writer

### ISSUE THREAD

**Issue by JuanFMontesinos** [lanpa/tensorboardX#345]
Hi
I've noticed (also after reading other issues) that add_text does not display it until you closes the writer.
tensorboardX 1.6
tensorboard 1.8
Is there any reason why?

**Comment by dumitrescustefan** [comment]
I have the same problem. Also, writing several times (with incremental time steps) only produces the first step on screen.

**Comment by BPiepmatz** [comment]
For me it is also not showing even though with writer.close()
this code produces the image below:
`writer.add_text('lr', 'learning rate', lr)
    writer.add_text('wd', 'weight decay', wd)`

<img width="1669" alt="screen shot 2019-02-15 at 17 11 03" src="https://user-images.githubusercontent.com/26856807/52869131-41ebfd00-3145-11e9-8ccb-efc8a1bbf438.png">


tensorboardX 1.6
pytorch 0.4
tensorboard 1.12

**Comment by lanpa** [comment]
Hi all, I need some code to reproduce the issue. @BPiepmatz The third argument accepts integer (the iteration number). For your case, `add_text('learning rate', str(lr), 1)` makes more sense.

**Comment by BPiepmatz** [comment]
@lanpa Oh I see, same issue still appearing. This is the whole code `    learning_rates = [1e-2, 1e-3, 1e-4, 1e-5, 1e-6, 1e-7, 1e-8]
    wd = 0.0002
    opt_names = ['RMSprop']
    nAveGrad = 5  # Average the gradient every nAveGrad iterations
    nEpochs = 200 * nAveGrad  # Number of epochs for train
    snapshot = nEpochs  # Store a model every snapshot epochs
    gpu_id = 0
    device = torch.device("cuda:" + str(gpu_id) if torch.cuda.is_available() else "cpu")

    for lr in learning_rates:
        for opt_name in opt_names:
            model_file_name = datetime.now(gettz('Europe/Berlin')).strftime(
                '%b%d_%H-%M-%S') + '_' + socket.gethostname() + '-' + str(
                lr) + '-' + str(nEpochs) + '-' + str(wd) + '-' + str(lr) + '-' + str(opt_name)
            log_dir = os.path.join('tensorboard_log/runs', model_file_name)

            # Logging into Tensorboard
            writer = SummaryWriter(log_dir=log_dir)

            # Load Network
            print('Loading model...')
            net = CustomNet(pretrained=False)
            optimizer = choose_optimizer(opt_name, net)
            writer.add_text('lr', str(lr), 1)
            writer.add_text('wd', str(wd), 2)
            # writer.add_text('nAveGrad', nAveGrad, 2)
            # writer.add_text('Optimizer', opt_name, 3)

            x = torch.randn(1, 3, 480, 854)
            y = torch.randn(1, 3, 480, 854)
            writer.add_graph(net, (x, y))
            net.to(device)
            writer.close()
            print('Model loaded')`

Does this help? 

Best

**Comment by lanpa** [comment]
@BPiepmatz It should be writer.add_text('lr', str(lr), n_iter)
I would write:
```
for n_iter, lr in enumerate(learning_rates):
  writer.add_text('lr', str(lr), n_iter)
```

**Comment by ZizhouJia** [comment]
I also have the same problem
tensorboardX 1.6
pytorch 1.0
tensorboard 1.12

**Comment by Guitaricet** [comment]
I have the very same problem - blank field instead of text.

After checking .tfevents file, everything seems to be on its place. It is hard to say definitely, but It is possible, that this is a tensorboard problem.

Part of .tfevents where I can find my texts:
```
loss/train^U/A  Cpç\^N"^@^@^@^@^@^@^@x^Q=ý      (>ò^H"1×A^PÐ^O*^T
^R
^Kspeed/train^U<89>×^HFMW^B3-^@^@^@^@^@^@^@<A<8a>û      2       û^H"1×A^PÐ^O*^_
^]
^Vgpu_memory_usage/train^U^@<88><9c>EÇd®t0^@^@^@^@^@^@^@<95>õÐ_ L3ô     "1×A^PÐ^O*"
 :^DtestB^N^H^G^R^D^R^B^H^AB^DtestJ^H
^F
^Dtext^P^FÙ2¯^B^@^@^@^@^@^@r¥öÔ ^P9ô    "1×A^PÐ^O* ^E
<9d>^E:^FsourceB<88>^E^H^G^R^D^R^B^H^ABý^DChavez checked his bolter and slapped home afresh magazine .
I am now married to Gweneth , who is English , and we have two children , Carl and Michelle .
In other words , NATO is very much interested in a number of joint operations with Russia .
The purpose in using such contracts and clauses is to provide the firm legal basis required for concluding an agreement acceptable to both sides .
Of course , political struggles are never - ending , but in this case the mass media are hostage to artificially created events that are often orchestrated by spin doctors .
By 2005 it became necessary to address additional user requirements .J^H
^F
^Dtexth²¢hÀ^D^@^@^@^@^@^@jú<9c>ö        )>ô     "1×A^PÐ^O*±
®       :^FtargetB<99>  ^H^G^R^D^R^B^H^AB<8e>   <s> Ð<9a>Ð°Ð¿Ð¸Ñ<82>Ð°Ð½ Ð¿Ñ<80>Ð¾Ð²ÐµÑ<80>Ð¸Ð» ÐµÐ³Ð¾ Ð±Ð¾Ð»Ñ<82>ÐµÑ<80> Ð¸ Ð²Ñ<81>Ñ<82>Ð°Ð²Ð¸Ð» Ð½Ð¾Ð²Ñ<8b>Ð¹ Ð¼Ð°Ð³Ð°Ð·Ð¸Ð½ . </s>
<s> Ð¡ÐµÐ¹Ñ<87>Ð°Ñ<81> Ñ<8f> Ð¶ÐµÐ½Ð°Ñ<82> Ð½Ð° Ð<93>Ð²Ð¸Ð½ÐµÑ<82> , Ð¾Ð½Ð° Ð°Ð½Ð³Ð»Ð¸Ñ<87>Ð°Ð½ÐºÐ° , Ð¸ Ñ<83> Ð½Ð°Ñ<81> Ð´Ð²Ð¾Ðµ Ð´ÐµÑ<82>ÐµÐ¹ : Ð<9a>Ð°Ñ<80>Ð» Ð¸ Ð<9c>Ð¸Ñ<88>ÐµÐ»Ñ<8c> . </s>
<s> Ð<94>Ñ<80>Ñ<83>Ð³Ð¸Ð¼Ð¸ Ñ<81>Ð»Ð¾Ð²Ð°Ð¼Ð¸ , Ð<9d>Ð<90>Ð¢Ð<9e> Ð²ÐµÑ<81>Ñ<8c>Ð¼Ð° Ð·Ð°Ð¸Ð½Ñ<82>ÐµÑ<80>ÐµÑ<81>Ð¾Ð²Ð°Ð½Ð¾ Ð² Ñ<80>Ñ<8f>Ð´Ðµ Ñ<81>Ð¾Ð²Ð¼ÐµÑ<81>Ñ<82>Ð½Ñ<8b>Ñ<85> Ñ<81> Ð Ð¾Ñ<81>Ñ<81>Ð¸ÐµÐ¹ Ð¾Ð¿ÐµÑ<80>Ð°Ñ<86>Ð¸Ð¹ , Ð²Ñ<8b>Ð¿Ð¾Ð»Ð½ÐµÐ½Ð¸Ðµ ÐºÐ¾Ñ<82>Ð¾Ñ<80>Ñ<8b>Ñ<85> Ð´Ð»Ñ<8f> Ð½Ð¸Ñ<85> Ð¶Ð¸Ð·Ð½ÐµÐ½Ð½Ð¾ Ð²Ð°Ð¶Ð½Ð¾ . </s>
<s> Ð¦ÐµÐ»Ñ<8c> Ð¸Ñ<81>Ð¿Ð¾Ð»Ñ<8c>Ð·Ð¾Ð²Ð°Ð½Ð¸Ñ<8f> Ñ<82>Ð°ÐºÐ¸Ñ<85> ÐºÐ¾Ð½Ñ<82>Ñ<80>Ð°ÐºÑ<82>Ð¾Ð² Ð¸ Ð¾Ð³Ð¾Ð²Ð¾Ñ<80>Ð¾Ðº - Ñ<8d>Ñ<82>Ð¾ Ð¾Ð±ÐµÑ<81>Ð¿ÐµÑ<87>ÐµÐ½Ð¸Ðµ Ð¿Ñ<80>Ð¾Ñ<87>Ð½Ð¾Ð¹ Ð¿Ñ<80>Ð°Ð²Ð¾Ð²Ð¾Ð¹ Ð±Ð°Ð·Ð¾Ð¹ , Ð½ÐµÐ¾Ð±Ñ<85>Ð¾Ð´Ð¸Ð¼Ð¾Ð¹ Ð´Ð»Ñ<8f> Ð·Ð°ÐºÐ»Ñ<8e>Ñ<87>ÐµÐ½Ð¸Ñ<8f> Ð¿Ñ<80>Ð¸ÐµÐ¼Ð»ÐµÐ¼Ð¾Ð³Ð¾ Ð´Ð»Ñ<8f> Ð¾Ð±ÐµÐ¸Ñ<85> Ñ<81>Ñ<82>Ð¾Ñ<80>Ð¾Ð½ Ñ<81>Ð¾Ð³Ð»Ð°Ñ<88>ÐµÐ½Ð¸Ñ<8f> . </s>
<s> Ð<9d>Ð¾ Ð¡Ð<9c>Ð<98> Ð² Ð´Ð°Ð½Ð½Ð¾Ð¼ Ñ<81>Ð»Ñ<83>Ñ<87>Ð°Ðµ Ñ<81>Ñ<82>Ð°Ð½Ð¾Ð²Ñ<8f>Ñ<82>Ñ<81>Ñ<8f> Ð·Ð°Ð»Ð¾Ð¶Ð½Ð¸ÐºÐ°Ð¼Ð¸ Ñ<81>Ð¾Ð·Ð´Ð°Ð½Ð½Ñ<8b>Ñ<85> Ñ<81>Ð¿ÐµÑ<86>Ð¸Ð°Ð»Ñ<8c>Ð½Ð¾ Ñ<81>Ð¾Ð±Ñ<8b>Ñ<82>Ð¸Ð¹ , Ð·Ð° ÐºÐ¾Ñ<82>Ð¾Ñ<80>Ñ<8b>Ð¼Ð¸ Ñ<87>Ð°Ñ<81>Ñ<82>Ð¾ Ñ<81>Ñ<82>Ð¾Ñ<8f>Ñ<82> Ð¿Ð¾Ð»Ð¸Ñ<82>Ñ<82>ÐµÑ<85>Ð½Ð¾Ð»Ð¾Ð³Ð¸ . </s>
<s> Ð<9a> 2005 Ð³Ð¾Ð´Ñ<83> Ð¿Ð¾Ñ<8f>Ð²Ð¸Ð»Ð°Ñ<81>Ñ<8c> Ð½ÐµÐ¾Ð±Ñ<85>Ð¾Ð´Ð¸Ð¼Ð¾Ñ<81>Ñ<82>Ñ<8c> Ð² Ñ<80>Ð°Ñ<81>Ñ<88>Ð¸Ñ<80>ÐµÐ½Ð¸Ð¸ Ð²Ð¾Ð·Ð¼Ð¾Ð¶Ð½Ð¾Ñ<81>Ñ<82>ÐµÐ¹ Ð¿Ñ<80>Ð¾Ñ<82>Ð¾ÐºÐ¾Ð»Ð° . </s>J^H
^F
^Dtext0wíOû^A^@^@^@^@^@^@^^<9f><98>û    ^T[ô    "1×A^PÐ^O*ì^C
é^C:
translatedBÐ^C^H^G^R^D^R^B^H^ABÅ^C<unk> <unk> Ð¸ <unk> <unk> .
Ð¯ <unk> , <unk> , <unk> , <unk> , <unk> , <unk> Ð¸ <unk> .
Ð<9a>Ñ<80>Ð¾Ð¼Ðµ Ñ<82>Ð¾Ð³Ð¾ , <unk> , <unk> , <unk> <unk> <unk> <unk> .
<unk> <unk> , <unk> Ð¸ <unk> , <unk> , <unk> Ð¸ <unk> , Ð² Ñ<82>Ð¾Ð¼ Ñ<87>Ð¸Ñ<81>Ð»Ðµ ÐºÐ°Ðº <unk> <unk> , <unk> <unk> <unk> <unk> .
<unk> <unk> , Ð½Ð¾ Ð½Ðµ <unk> , Ð½Ð¾ Ð½Ðµ <unk> , Ð½Ð¾ Ð½Ðµ <unk> , Ð½Ð¾ Ð² Ñ<8d>Ñ<82>Ð¾Ð¼ Ñ<81>Ð»Ñ<83>Ñ<87>Ð°Ðµ <unk> <unk> .
Ð<92> Ð½Ð°Ñ<81>Ñ<82>Ð¾Ñ<8f>Ñ<89>ÐµÐµ Ð²Ñ<80>ÐµÐ¼Ñ<8f> <unk> <unk> <unk> <unk> <unk> .J^H
^F
^Dtext13ûx!^@^@^@^@^@^@^@^A{ìö  uèj
"1×A^PÚ^O*^S
^Q

loss/train^UÊ<82>^OC´æ+."^@^@^@^@^@^@^@x^Q=ý    cêj
"1×A^PÚ^O*^T
^R
^Kspeed/train^U
^P¨DÞ>"^R!^@^@^@^@^@^@^@^A{ìö   a¼¢
"1×A^Pä^O*^S
^Q
```

**Comment by lanpa** [comment]
We will have flush() for the summarywriter after #451 is merged. Then you can call writer.flush() manually to replace `close()`.
